### PR TITLE
Add controller stub and circle CI configuation

### PR DESCRIPTION
**Pull request by skade**
New version of #65, checking if CircleCI runs.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,6 +32,24 @@ jobs:
...
+  tests_rust:
+    docker:
+      - image: circleci/rust:1.3.7
+    steps:
+      - checkout
+
+      - run:
+          name: install dependencies
+          command: |
+              rustup install nightly
```
Comment (tanertopal) [inline .circleci/config.yml]: Is it ok if we use a fixed version of rust instead of the nightly or do you see any advantages in using the nightly?

Comment (skade) [reply .circleci/config.yml]: Yes, specifically `async/await`. It's 8 weeks out and I wouldn't start a networked codebase without it nowadays. We will be able to switch to beta by September 26.

Comment (tanertopal) [reply .circleci/config.yml]: Ok, cool. Lets fix the version as soon as there is a stable rust version with `async/await`

**Comment by tanertopal** [comment]
Feel free to merge

File: .circleci/config.yml (hunk 1 above)
Comment (matklad) [reply .circleci/config.yml]: It might be a good idea to pin to a specific nightly though

Comment (skade) [reply .circleci/config.yml]: Right. Added a rust-toolchain file and skipped the install step.

**Pull request merged**